core/inflation/src/lib.rs

```diff
@@ -65,7 +66,7 @@ fn inflate(x: Balance, years: u8) -> Option<Balance> {
 	let ninety_nine = I95F33::from_num(99_u8) / 100_i128;
 	let pow = transcendental::pow::<I95F33, I95F33>(ninety_nine, sqrt).ok()?;
 	let ratio = I95F33::from_num(1_u8) - pow;
-	let inflation = I95F33::from_num(x) * ratio;
+	let inflation = I95F33::checked_from_num(x)? * ratio;
```

Comment:
What will happen to the mark `?` here? return `None`?

What will happen if it returns `None` here?

Reply:
What's the max value of  I95F33, it there possibility of overflow, just in case it was designed to utilise overflow

Reply:
Downstream usages:
- https://github.com/darwinia-network/darwinia/blob/9f3266b6af020edbce8c16b9ece688c7b79bad75/pallet/staking/src/lib.rs#L818-L826

Reply:
> What's the max value of I95F33, it there possibility of overflow, just in case it was designed to utilise overflow

Max value is 19807040628566084398385987584, 1.98 * 10 ^28, max value of ring balance could be 100 * 10^8 * 10^18, for the case of ring, it is not likely to cause a result None here.

Reply:
See the comment here: https://github.com/darwinia-network/darwinia/blob/9f3266b6af020edbce8c16b9ece688c7b79bad75/core/inflation/src/lib.rs#L62